### PR TITLE
TECS refactor to respect vehicle limits and avoid unnecessary state

### PR DESCRIPTION
**Pull request by dagar**
When reviewing https://github.com/PX4/ecl/pull/467 I noticed a few things in TECS.

* vehicle limits weren't being respected with the load factor correction
* the airspeed less mode is hard to follow because it goes through all the same integrator logic, but then ignores the result
* the throttle slew rate was also only respected in non-climbout airspeed mode
* it's not necessary to store the previous throttle setpoint.

@philipoe would you mind reviewing?

### CODE REVIEW

**Comment by CarlOlsson** [comment]
@dagar is this waiting for something to get in?

**Comment by dagar** [comment]
> @dagar is this waiting for something to get in?

Testing PX4 side. I'll also rebase it to current master.

**Comment by dagar** [comment]
Auto created PX4/Firmware branch containing this change for testing. https://github.com/PX4/Firmware/tree/ecl_PR-468

@CarlOlsson an additional review also couldn't hurt.

**Comment by CarlOlsson** [comment]
> @CarlOlsson an additional review also couldn't hurt.

Will do it on Monday!

File: tecs/tecs.cpp [hunk 1]
```diff
@@ -339,61 +350,58 @@ void TECS::_update_throttle_setpoint(const float throttle_cruise, const matrix::
...
-		_last_throttle_setpoint = _throttle_setpoint;
+			// Calculate gain scaler from specific energy error to throttle
+			const float STE_to_throttle = 1.0f / (_throttle_time_constant * (_STE_rate_max - _STE_rate_min));
 
-		if (_integrator_gain > 0.0f) {
-			// Calculate throttle integrator state upper and lower limits with allowance for
-			// 10% throttle saturation to accommodate noise on the demand.
-			float integ_state_max = _throttle_setpoint_max - _throttle_setpoint + 0.1f;
-			float integ_state_min = _throttle_setpoint_min - _throttle_setpoint - 0.1f;
+			// Add proportional and derivative control feedback to the predicted throttle
```
Comment (CarlOlsson) [inline tecs/tecs.cpp]:
```suggestion
			// Add proportional and derivative control feedback to the throttle setpoint
```

**Comment by priseborough** [comment]
@CarlOlsson and @bresch What does @dagar need to do to get this over the line?

**Comment by CarlOlsson** [comment]
From my side it is good to go, but I guess answering the above questions regarding the differences to master would be beneficial for users flying without airspeed sensor

**Comment by bresch** [comment]
@dagar Did we get some tests from the flight test team? It would be great to merge that PR now.

**Comment by priseborough** [comment]
@dagar I don't this going to waste. Was there any further testing?

**Comment by priseborough** [comment]
Closing this because questions wrt impact on non-airspeed users and testing remain unanswered.
